presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -275,38 +279,82 @@ private void doTestHiveCoercion(HiveTableDefinition tableDefinition)
-                        ImmutableMap.of(-2, asMap("ti2bi", null, "int2bi", -2323L, "float2double", -1.5, "add", null)),
-                        1));
+        List<Row> expectedRows;
+        Connection connection = defaultQueryExecutor().getConnection();
+        if (usingPrestoJdbcDriver(connection)) {
+            expectedRows = ImmutableList.of(
+                        row(
+                                -1,
+                                2,
+                                -3L,
+                                100,
+                                -101L,
+                                2323L,
+                                "12345",
+                                0.5,
+                                asMap("keep", "as is", "ti2si", (short) -1, "si2int", 100, "int2bi", 2323L, "bi2vc", "12345"),
+                                ImmutableList.of(asMap("ti2int", 2, "si2bi", -101L, "bi2vc", "12345")),
+                                asMap(2, asMap("ti2bi", -3L, "int2bi", 2323L, "float2double", 0.5, "add", null)),
+                                1),
+                        row(
+                                1,
+                                -2,
+                                null,
+                                -100,
+                                101L,
+                                -2323L,
+                                "-12345",
+                                -1.5,
+                                asMap("keep", null, "ti2si", (short) 1, "si2int", -100, "int2bi", -2323L, "bi2vc", "-12345"),
+                                ImmutableList.of(asMap("ti2int", -2, "si2bi", 101L, "bi2vc", "-12345")),
+                                ImmutableMap.of(-2, asMap("ti2bi", null, "int2bi", -2323L, "float2double", -1.5, "add", null)),
+                                1));
+        }
+        else if (usingTeradataJdbcDriver(connection)) {
+            expectedRows = ImmutableList.of(
+                    row(
+                            -1,
+                            2,
+                            -3L,
+                            100,
+                            -101L,
+                            2323L,
+                            "12345",
+                            0.5,
+                            "[\"as is\",-1,100,2323,\"12345\"]",
+                            "[[2,-101,\"12345\"]]",
+                            "{\"2\":[-3,2323,0.5,null]}",
+                            1),
+                    row(
+                            1,
+                            -2,
+                            null,
+                            -100,
+                            101L,
+                            -2323L,
+                            "-12345",
+                            -1.5,
+                            "[null,1,-100,-2323,\"-12345\"]",
+                            "[[-2,101,\"-12345\"]]",
+                            "{\"-2\":[null,-2323,-1.5,null]}",
```

Comment:
Is it a regression / backward incompatible change in the driver?

Reply:
I do not know if it was ever supported, we do not run simba tests against master.